src/js/index.js

```diff
@@ -0,0 +1,79 @@
+const $ = (selector) => document.querySelector(selector);
+  
+function App() {
+
+  const createMenuName = () => {
+
+    if($("#espresso-menu-name").value === ""){
+      alert("값을 입력해주세요");
+      return;
+    }
+
+      const espressoMenuName = $("#espresso-menu-name").value;
+      
+      $("#espresso-menu-list").insertAdjacentHTML("beforeend", menuItemTemplate(espressoMenuName));
+      updateMenuCount();
+      $("#espresso-menu-name").value = "";
+  };
+  
+  const updateMenuCount = () => {
+    const menuCount = $("#espresso-menu-list").querySelectorAll('li').length;
+    $(".menu-count").innerText = `총 ${menuCount} 개`
+  };
+
+  const menuItemTemplate = (espressoMenuName) => {
+    return `<li class="menu-list-item d-flex items-center py-2">
+    <span class="w-100 pl-2 menu-name">${espressoMenuName}</span>
+    <button
+      type="button"
+      class="bg-gray-50 text-gray-500 text-sm mr-1 menu-edit-button"
+    >
+      수정
+    </button>
+    <button
+      type="button"
+      class="bg-gray-50 text-gray-500 text-sm menu-remove-button"
+    >
+      삭제
+    </button>
+  </li>`
+  };
+
+  const removeMenuName = (e) => {
+    e.target.closest("li").remove();
+    updateMenuCount(); 
+  };
+
+  const editMenuName = (e) =>{
+    const $menuName = e.target.closest("li").querySelector(".menu-name");
+    const editedName = prompt("메뉴명을 수정하세요", $menuName.innerText);
+    $menuName.innerText = editedName;
+  };
+
+  $("#espresso-menu-list").addEventListener("click", (e) => {
+    if (e.target.classList.contains("menu-edit-button")){
+      editMenuName(e);
+    }
+
+    if(e.target.classList.contains("menu-remove-button")){
+      if(confirm("정말 삭제하시겠습니까?")){
+        removeMenuName(e);
+      }
+    }
+  });
+
+  // form 태그가 자동으로 전송되는 걸 막기
+  $("#espresso-menu-form").addEventListener("submit", (e) => {
+    e.preventDefault();
+  })
+  
+  // 메뉴의 이름을 입력 받기
+  $("#espresso-menu-name").addEventListener("keydown", (e) => {
+    if(e.key !== "Enter") return;
+    createMenuName();
+    });
+  
+  $("#espresso-menu-submit-button").addEventListener("click", createMenuName);
```

Comment:
`submit` 이벤트에 `createMenuName` 함수를 넣어주면 더 간결하지 않을까요?

Reply:
submit이벤트에서 createMenuName를 호출하는것을 말씀하시는 것일까요? :)
submit 이벤트 부분은 form 태그 때문에 자동으로 새로고침되는 부분을 막기위해서 e.preventDefault(); 만 넣고 입력받는 이벤트는 아래의 keydown과 click에서만 createMenuName을 호출하고 있어서~ 호출할 필요는 없을 것 같습니다~

Reply:
```suggestion
  // form 태그가 자동으로 전송되는 걸 막기
  $("#espresso-menu-form").addEventListener("submit", (e) => {
    e.preventDefault();
    createMenuName();
  })
```
아아 동작은 동일하지만 이렇게 줄이면 코드가 조금 더 간결하지 않을까 하는 의미였습니다 !

Reply:
form 요소 안의 input에 Enter keydown이벤트를 받는 방식을, 위에 106님이 하신 것처럼 할 수 있습니다
ref: https://codesandbox.io/s/input-in-form-2o9x9i

는 이미 달아주셧군요